src/par_iter/mod.rs

```diff
@@ -363,6 +365,16 @@ pub trait ParallelIterator: Sized {
         reduce(self, reduce_op)
     }
 
+    /// Takes two iterators and creates a new iterator over both.
+    ///
+    /// As with most parallel iterators, the order in which these are
+    /// combined is completely unspecified.
```

Comment:
Hmm. I am not sure how to interpret this comment. I would expect (as a user) that if you do `a.chain(b)` then you get `a` first and then `b` -- e.g., if we use `enumerate()`. Is this not true? (I think it is, from the impl)